perf/perf_24x7_counter.py

```diff
@@ -0,0 +1,230 @@
+        if "Performance counter stats for 'CPU(s) 0'" not in result1.stderr:
+            self.nfail += 1
+            self.log.info('FAIL : perf unable to recognize domain name\
+                          in param=value format')
+        else:
+            self.log.info('perf recognized domain name in param=value format')
+
+    def check_domain_not_existing(self):
+        result1 = self.event_stat(',domain=12,core=1/ sleep 1')
+        if "not supported" not in result1.stderr:
+            self.nfail += 1
+            self.log.info('FAIL : domain does not exist but perf listed\
+                          has supported')
+        else:
+            self.log.info('perf listed non-existing domain as unsupported')
+
+    def check_all_domains(self):
+        for domain in range(1, 6):
+            result1 = self.event_stat(',domain=%s,core=1/ sleep 1' % domain)
+            if "Performance counter stats for 'CPU(s)\
+                    0'" not in result1.stderr:
+                self.nfail += 1
+                self.log.info('FAIL : perf unable to recognize domain name in\
+                              param=value format for all domains')
+            else:
+                self.log.info('perf recognized domain name in param=value\
+                              format for all 6 domains')
+
+    def event_w_chip_param(self):
+        event_out = process.run("cat\
+            %s/events/PM_XLINK_CYCLES" % self.event_sysfs)
+        if "chip=?" in event_out.stdout:
+            self.log.info('sysfs entry has chip entry')
+        else:
+            self.nfail += 1
+            self.log.info('FAIL : sysfs does not have chip entry')
+
+        if os.path.exists("%s/format/chip" % self.event_sysfs):
+            self.log.info('chip file exists')
+        else:
+            self.nfail += 1
+            self.log.info('FAIL : chip file does not exist')
+
+    def event_wo_chip_param(self):
+        output_chip_missing = self.event_stat1('hv_24x7/PM_XLINK_CYCLES,\
+            domain = 1/ /bin/true')
+        if "Required parameter 'chip' not\
+                specified" not in output_chip_missing.stdout:
+            self.nfail += 1
+            self.log.info('FAIL : perf unable to detect chip\
+                          parameter missing')
+        else:
+            self.log.info('perf detected chip parameter missing')
+        output_chip = self.event_stat1('hv_24x7/PM_XLINK_CYCLES,\
+                                       domain = 1,chip = 1/ /bin/true')
+        if "Performance counter stats for" not in output_chip.stderr:
+            self.nfail += 1
+            self.log.info('FAIL : performance counter stats for missing')
+
+    #Helper functions
+    def event_helper(self, event):
+        search_suffix = process.run('ls %s/events |grep -E\
+            %s' % (self.event_sysfs, event), ignore_status=True)
+        if search_suffix.stdout:
+            self.nfail += 1
+            self.log.info('FAIL : Found %s  suffixes in event name' % event)
```

Comment:
Here, I check VCPU/ PHYS_CORE . It is expected that this event does not exist. if they exist then it is fail.